spring-integration-core/src/main/java/org/springframework/integration/dsl/DelayerEndpointSpec.java

```diff
@@ -97,6 +104,110 @@ public DelayerEndpointSpec delayExpression(String delayExpression) {
+	 * complete. If the error flow throws an exception, the release will be re-attempted.
+	 * If there is a transaction advice on the release task, the error flow is called
+	 * within the transaction.
+	 * @param channel the channel name.
+	 * @return the endpoint spec.
+	 * @see #maxAttempts(int)
+	 * @see #retryDelay(long)
+	 * @since 5.0.8
+	 */
+	public DelayerEndpointSpec delayedMessageErrorChannel(String channel) {
+		this.handler.setDelayedMessageErrorChannelName(channel);
+		return this;
+	}
+
+	/**
+	 * Set the maximum number of release attempts for when message release fails.
+	 * Default {@value DelayHandler#DEFAULT_MAX_ATTEMPTS}.
+	 * @param maxAttempts the max attempts.
+	 * @return the endpoint spec.
+	 * @see #retryDelay(long)
+	 * @since 5.0.8
+	 */
+	public DelayerEndpointSpec maxAttempts(int maxAttempts) {
+		this.handler.setMaxAttempts(maxAttempts);
+		return this;
+	}
+
+	/**
+	 * Set an additional delay to apply when retrying after a release failure.
+	 * Default {@value DelayHandler#DEFAULT_RETRY_DELAY}.
+	 * @param retryDelay the retry delay.
+	 * @return the endpoint spec.
+	 * @see #maxAttempts(int)
+	 * @since 5.0.8
+	 */
+	public DelayerEndpointSpec retryDelay(long retryDelay) {
+		this.handler.setRetryDelay(retryDelay);
+		return this;
+	}
+
+	/**
+	 * Specify a {@link TransactionInterceptor} {@link Advice} with default
+	 * {@code PlatformTransactionManager} and {@link DefaultTransactionAttribute} for the
+	 * {@link MessageHandler}.
+	 * @return the spec.
+	 * @since 5.0.8
+	 */
+	public DelayerEndpointSpec transactionalRelease() {
+		TransactionInterceptor transactionInterceptor = new TransactionInterceptorBuilder().build();
+		this.componentsToRegister.put(transactionInterceptor, null);
+		return delayedAdvice(transactionInterceptor);
+	}
+
+	/**
+	 * Specify a {@link TransactionInterceptor} {@link Advice} for the {@link MessageHandler}.
+	 * @param transactionInterceptor the {@link TransactionInterceptor} to use.
+	 * @return the spec.
+	 * @see TransactionInterceptorBuilder
+	 * @since 5.0.8
+	 */
+	public DelayerEndpointSpec transactionalRelease(TransactionInterceptor transactionInterceptor) {
+		return delayedAdvice(transactionInterceptor);
+	}
+
+	/**
+	 * Specify a {@link TransactionInterceptor} {@link Advice} with the provided
+	 * {@code PlatformTransactionManager} and default {@link DefaultTransactionAttribute}
+	 * for the {@link MessageHandler}.
+	 * @param transactionManager the {@link PlatformTransactionManager} to use.
+	 * @param handleMessageAdvice the flag to indicate the target {@link Advice} type:
+	 * {@code false} - regular {@link TransactionInterceptor}; {@code true} -
+	 * {@link org.springframework.integration.transaction.TransactionHandleMessageAdvice}
+	 * extension.
+	 * @return the spec.
+	 * @since 5.0.8
+	 */
+	public DelayerEndpointSpec transactionalRelease(PlatformTransactionManager transactionManager,
+			boolean handleMessageAdvice) {
```

Comment:
Why do we need this variant of the `TransactionInterceptor` if we are going to proxy only a `DelayHandler.ReleaseMessageHandler` anyway?

Reply:
Good catch.